siliconcompiler/core.py

```diff
@@ -4070,25 +3983,33 @@ def _reset_flow_nodes(self, flow, steplist, indexlist):
                     all_indices_failed = False
 
             if should_resume and all_indices_failed:
-                # When running with -resume, we re-run any step in steplist that
+                # When running with -resume, we re-run any step in flowgraph that
                 # had all indices fail.
                 for index in self.getkeys('flowgraph', flow, step):
-                    if (step, index) in node_list:
+                    if (step, index) in nodes_to_execute:
                         self.set('flowgraph', flow, step, index, 'status', None)
                         for metric in self.getkeys('metric'):
                             self._clear_metric(step, index, metric)
                         for record in self.getkeys('record'):
                             self._clear_record(step, index, record)
 
-    def _prepare_nodes(self, nodes_to_run, processes, flow, status, steplist, indexlist):
+    def clean_build_dir(self):
+        if not self.get('option', 'resume') and not self.get('arg', 'step') \
+                and not self.get('option', 'from') and not self.get('record', 'remoteid'):
```

Comment:
can you make a ticket to track the remoteid thing? I'm not 100% sure this is correct, but I can't think of how it is wrong.

Reply:
SC-636